### PR TITLE
Fix build process

### PR DESCRIPTION
**Pull request by DanGrayson**

### CODE REVIEW

File: .github/workflows/build-libs.yml [hunk 1]
```diff
@@ -30,29 +30,27 @@ jobs:
...
         run: |
-          pushd $GITHUB_WORKSPACE
-          make build-coq
           make -j2
-          make install
-          export PATH=$PATH:$PWD/sub/coq/bin/
-          echo $PATH
-          popd
-          # build TypeTheory
+	  : set the Coq path
```
Comment (benediktahrens) [inline .github/workflows/build-libs.yml:41]: This looks like a tab instead of space, which is a syntax error in this kind of file.

**Comment by DanGrayson** [comment]
@benediktahrens -- why does it say above "Build-UniMath-and-Sanity-Checks Expected" ??  I no longer have a step with that name.

**Comment by benediktahrens** [comment]
> @benediktahrens -- why does it say below "Build-UniMath-and-Sanity-Checks Expected" ?? I no longer have a step with that name.

Because that is the CI job that is marked as "required" before merging is allowed - this is a setting in https://github.com/UniMath/UniMath/settings/branches. Once the new CI jobs are set up, I will update that setting. For now, just ignore this expectation.

**Comment by DanGrayson** [comment]
@benediktahrens -- it looks ready to be merged, to me.  Do you see any further problems?

**Comment by benediktahrens** [comment]
> @benediktahrens -- it looks ready to be merged, to me. Do you see any further problems?

Great, thanks. 

Does `./UniMath/.dir-locals.el` need to be adapted? There is the line
```
	     (setq coq-prog-name (concat unimath-topdir "sub/coq/bin/coqtop"))
```
which might be superfluous or even incorrect if Coq is installed elsewhere.

**Comment by DanGrayson** [comment]
> > @benediktahrens -- it looks ready to be merged, to me. Do you see any further problems?
> 
> Great, thanks.
> 
> Does `./UniMath/.dir-locals.el` need to be adapted? There is the line
> 
> ```
> 	     (setq coq-prog-name (concat unimath-topdir "sub/coq/bin/coqtop"))
> ```
> 
> which might be superfluous or even incorrect if Coq is installed elsewhere.

Ah, yes.  That's important to fix.  I had it in mind but then forgot all about it.

**Review by nmvdw — COMMENTED** [review]
For me:
- The new building process works. I am able to go through files afterwards and I can use it elsewhere 
- I think the installation instructions are clear (it is how I installed UniMath before).

**Comment by peterlefanulumsdaine** [comment]
Thanks very much indeed @DanGrayson !  I was just reviewing this when it got merged; the main thing I was going to say was to confirm that it works smoothly for me (on Mac OS Monterey 12.3.1) and looks great, but I also had a few small tweaks to suggest, so I’ve started implementing those in a followup PR, #1496.

**Comment by DanGrayson** [comment]
Thanks, any improvements are welcome!

**Comment by DanGrayson** [comment]
That seems to work still:
```bash
make -j1 UniMath/Foundations/Init.vo
--- making .coq_makefile_input
coq_makefile -f .coq_makefile_input -o .coq_makefile_output
mv .coq_makefile_output build/CoqMakefile.make
make: Nothing to be done for 'UniMath/Foundations/Init.vo'.
```

**Comment by rmatthes** [comment]
This seems only to work since nothing has to be done. If you delete the .vo file, it is different:
```
make -j1 UniMath/Bicategories/MonoidalCategories/ActionBasedStrongFunctorsWhiskeredMonoidal.vo
make: rien à faire pour « UniMath/Bicategories/MonoidalCategories/ActionBasedStrongFunctorsWhiskeredMonoidal.vo ».
rm UniMath/Bicategories/MonoidalCategories/ActionBasedStrongFunctorsWhiskeredMonoidal.vo
make -j1 UniMath/Bicategories/MonoidalCategories/ActionBasedStrongFunctorsWhiskeredMonoidal.vo
make: *** Aucune règle pour fabriquer la cible « UniMath/Bicategories/MonoidalCategories/ActionBasedStrongFunctorsWhiskeredMonoidal.vo ». Arrêt.
```
Sorry for the French locale.

**Comment by DanGrayson** [comment]
Oh, right, of course.  You could add `-f build/CoqMakefile.make` to the command line -- that seems to work.

**Comment by rmatthes** [comment]
Thank you - this is precious information that I add to my list of hints for interaction with UniMath. It would be good to have it in the documentation somewhere, too.

**Comment by DanGrayson** [comment]
I've made the old way work in commit 3c017b862aff1e776ab46dd16baca384d88cc6b4

**Comment by DanGrayson** [comment]
... so maybe we don't have to document this work-around.

**Pull request merged**